core/src/main/scala/kafka/log/remote/RemoteLogManager.scala

```diff
@@ -125,7 +125,7 @@ class RemoteLogManager(fetchLog: TopicPartition => Option[Log],
   private val rlmScheduledThreadPool = new RLMScheduledThreadPool(poolSize)
 
   // topic ids received on leadership changes
-  @volatile private var topicPartitionIds: Map[TopicPartition, TopicIdPartition] = Map.empty
+  private val topicIds: mutable.Map[String, Uuid] = mutable.Map.empty
```

Comment:
Having these at the partition level will not create transient `TopicIdPartition` objects but it always keeps them. We can look at this later.